src/components/Modal.mock.ts

```diff
@@ -0,0 +1,29 @@
+import { FunctionComponent } from 'react'
+
+
+export type ModalExports = typeof import('src/components/Modal') & { __esModule: true }
+
+/**
+ * provides a mocked version of Modal module's Modal component that avoids
+ * pitfalls specific to missing services in jest simulated browser environment
+ * including dom measurement calls.
+ */
+export const mockModalModule = (): ModalExports => {
+  const originalModule = jest.requireActual<ModalExports>('src/components/Modal')
```

Comment:
note:
if a shared mock using this pattern needs to pull in secondary modules, it must use jest.requireActual() to pull them in.
This is because these helpers are expected to be called within a jest.mock() call, which jest is hoisting to the top of the test file.